src/main/java/ru/mingun/kaitai/struct/tree/ListNode.java

```diff
@@ -91,7 +94,21 @@ private List<ChunkNode> init() {
         try {
           final int s = arrStart.get(index);
           final int e = arrEnd.get(index);
-          children.add(create("[" + index + ']', obj, 0, s, e));
+
+          /*
+           We cannot access the generic type of the List at runtime due to
+           type erasure. Instead, look for a non-null element in the List,
+           which will tell us the type of the whole list.
+           */
+          final Class<?> valueClass;
+          final Optional<?> nonNullElement = value.stream().filter(Objects::nonNull).findAny();
+          if (nonNullElement.isPresent()) {
+            valueClass = nonNullElement.get().getClass();
+          } else {
+            valueClass = null;
```

Comment:
I think it is better to avoid `null` in that field:
```suggestion
            valueClass = Object.class;
```